### PR TITLE
Setup the frontend only when there is an appserver.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/833/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -2542,6 +2542,27 @@ def add_appserver_to_haproxy(app_id, ip, port, secret)
       return NOT_READY
     end
 
+    if !Nginx.is_app_already_configured(app_id)
+      begin
+        static_handlers = HelperFunctions.parse_static_data(app_id)
+        Djinn.log_run("chmod -R +r #{HelperFunctions.get_cache_path(app_id)}")
+      rescue Exception => e
+        # This specific exception may be a json parse error
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Add '.'.

File: AppController/djinn.rb [hunk 2]
```diff
@@ -2542,6 +2542,27 @@ def add_appserver_to_haproxy(app_id, ip, port, secret)
...
+        error_msg = "ERROR: Unable to parse app.yaml file for #{app_id}. "\
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Should we log the error here?

File: AppController/djinn.rb [hunk 3]
```diff
@@ -2542,6 +2542,27 @@ def add_appserver_to_haproxy(app_id, ip, port, secret)
...
+        static_handlers = []
+      end
+
+      Nginx.write_fullproxy_app_config(app_id,
+        @app_info_map[app_id]['nginx'],
+        @app_info_map[app_id]['nginx_https'], my_node.public_ip,
+        my_node.private_ip, @app_info_map[app_id]['haproxy'],
+        static_handlers, get_login.private_ip,
+        @app_info_map[app_id]['language'])
+      Djinn.log_info("Done setting fullproxy for application #{app}.")
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Add missing space.

File: AppController/lib/nginx.rb [hunk 4]
```diff
@@ -486,6 +486,22 @@ def self.write_fullproxy_app_config(app_name, http_port, https_port,
     return reload_nginx(config_path, app_name)
   end
 
+  # This function checks if Nginx has already configured the specified
+  # application.
+  #
+  # Args:
+  #  app_name: the application to check for.
```
Comment (menivaitsi) [inline AppController/lib/nginx.rb]: the -> The

File: AppController/djinn.rb (hunk 2 above)
Comment (obino) [inline AppController/djinn.rb]: place_error_app already has an error message. It would be double I think.

**Comment by obino** [comment]
The branch is in pending state now: we need to review the user experience before changing the behavior of the serer side.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1013/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1017/

File: AppController/djinn.rb [hunk 5]
```diff
@@ -2558,6 +2558,27 @@ def add_appserver_to_haproxy(app_id, ip, port, secret)
       return NOT_READY
     end
 
+    if !Nginx.is_app_already_configured(app_id)
+      begin
+        static_handlers = HelperFunctions.parse_static_data(app_id)
+        Djinn.log_run("chmod -R +r #{HelperFunctions.get_cache_path(app_id)}")
+      rescue Exception => e
+        # This specific exception may be a json parse error.
```
Comment (menivaitsi) [inline AppController/djinn.rb]: json -> JSON

File: AppController/djinn.rb [hunk 6]
```diff
@@ -4768,15 +4789,15 @@ def setup_appengine_application(app, state)
...
           begin
-            pid = app_manager.start_app(app, appengine_port,
+            pid = Integer(app_manager.start_app(app, appengine_port,
               get_load_balancer_ip(), app_language, xmpp_ip,
               HelperFunctions.get_app_env_vars(app),
-              Integer(@options['max_memory']), get_login.private_ip)
-          rescue FailedNodeException
+              Integer(@options['max_memory']), get_login.private_ip))
+          rescue FailedNodeException, ArgumentError
             Djinn.log_warn("Failed to talk to AppManager to start #{app}.")
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Should we be printing out the error?

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1019/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1022/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1024/

**Comment by cdonati** [comment]
Precise: https://ocd.appscale.com:8080/job/Daily%20Build/1027/

**Pull request merged**